### PR TITLE
RAJA Exercises

### PR DESCRIPTION
**Pull request by artv3**
I converted a 8 of the tutorial examples into hands on exercises.

1. ex_1_add-vectors.cpp		 
2. ex_2_dot-product.cpp	 
3. ex_3_indexset-segments.cpp
4. ex_4_atomic-binning.cpp
5. ex_5_batched-matrix-multiply.cpp 
6. ex_6_matrix-multiply.cpp
7. ex_7_tiled-matrix-transpose.cpp
8. ex_8_matrix-transpose-local-array.cpp

Currently they are in a different folder than the tutorial examples, but I can move them. 
These exercises require the students to add the RAJA variants.

### CODE REVIEW

File: exercises/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# All rights reserved.
+#
+# This file is part of RAJA.
+#
+# For details about use and distribution, please read RAJA/LICENSE.
+#
+###############################################################################
+
+raja_add_executable(
+  NAME tut_1_add-vectors
```
Comment (davidbeckingsale) [inline exercises/CMakeLists.txt]: Probably want to rename all these like `ex_1_add-vectors` or similar.

File: exercises/CMakeLists.txt~ [hunk 2]
```diff
@@ -0,0 +1,47 @@
+###############################################################################
```
Comment (davidbeckingsale) [inline exercises/CMakeLists.txt~]: This file shouldn't have been committed.

File: exercises/tut_1_add-vectors.cpp [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+//printResult(c, N);
+
+
+//----------------------------------------------------------------------------//
+// RAJA::seq_exec policy enforces strictly sequential execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA sequential vector addition...\n";
+
+  //EDIT: RAJA variant using RAJA::seq_exec
```
Comment (davidbeckingsale) [inline exercises/tut_1_add-vectors.cpp]: I would use "TODO" rather than "EDIT" here.

Comment (artv3) [reply exercises/tut_1_add-vectors.cpp]: Thanks, fixed.

File: exercises/CMakeLists.txt~ (hunk 2 above)
Comment (artv3) [reply exercises/CMakeLists.txt~]: Thanks, fixed.

**Pull request merged**